docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-narayana-lra</artifactId>
+    </dependency>
+----
+
+If there is a running coordinator then this is all you need in order to create
+new LRAs and to enlist participants with them.
+
+The LRA extension can be configured by placing an `application.properties` file
+in the `src/main/resources` directory. The only LRA specific property is
+`quarkus.lra.coordinator-url=<url>` which specifies the HTTP endpoint of an external
+coordinator, for example:
+
+[source,bash]
+----
+quarkus.lra.coordinator-url=http://localhost:8080/lra-coordinator
+----
+
+For a narayana coordinator the path component of the url is normally `lra-coordinator`.
+Coordinators can be obtained from `https://hub.docker.com/r/jbosstm/lra-coordinator`
+or you can build your own coordinator using a maven pom that includes the appropriate
+dependencies. A quarkus quickstart will be provided to show how to do this or you can
+take a look at one of the https://github.com/jbosstm/quickstart/tree/master/rts/lra-examples/lra-coordinator[narayana quickstarts].
+Another option would be to run it managed inside a WildFly application server.
+
+== Handling failures
+
+When an LRA is told to finish, i.e. when a method annotated with `@LRA(end = true, ...)`
+is invoked, the coordinator will instruct all services involved in the interaction to
+finish. If a service is unavailable (or still finishing) then the coordinator will retry
+periodically. It is the users responsibility to restart failed services on the same
+endpoint that they used when they first joined the LRA, or to tell the coordinator that
+they wish to be notified on new endpoints. An LRA is not deemed finished until *all*
+participants have acknowledged that they have finished.
+
+The coordinator is responsible for reliably creating and ending LRAs and for managing
+participant enlistment and it therefore must be available (for example if it or the
+network fail then something in the environment is responsible for restarting
+the coordinator or for repairing the network, respectively). To fulfil this task the
+coordinator must have access to durable storage for its logs (via a filesystem or in
+a database).  At the time of writing, managing coordinators is the responsibility of
+the user. An "out-of-the-box" solution will be forthcoming.
+
+== Examples
+
+The following is a simple example
+(https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc#java-annotations[taken from the specification document])
+of how to start an LRA and how to receive a notification when the LRA is later
+cancelled (the `@Compensate` annotated method is called) or closed (`@Complete` is called):
+
+[source,java]
+----
+@Path("/")
+@ApplicationScoped
+public class SimpleLRAParticipant
+{
+    @LRA(LRA.Type.REQUIRES_NEW)
+    @Path("/work")
+    @PUT
+    public void doInTransaction(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId)
```

Comment:
I know that transaction and LRA are close, but, it would be better to avoid any potential confusion.

Reply:
An LRA is a transaction, although not full ACID because we relax C and I, but it is still a transaction.

Reply:
You have provided a lot of useful comments, many of which require a little bit of dev work so I will go through all those first. When I've done that I will provide responses/resolutions to you comments.

For this particular comment, I will rename it `doInLongRunningAction` and include a comment that makes it clear that an LRA is not full ACID.

Reply:
+1